risc0/circuit/rv32im/src/execute/poseidon2.rs

```diff
@@ -79,89 +79,141 @@ impl Poseidon2State {
         }
     }
 
-    fn step(
+    fn step<C>(
         &mut self,
-        ctx: &mut impl Risc0Context,
+        ctx: &mut C,
         cur_state: &mut CycleState,
         next_state: CycleState,
         sub_state: u32,
-    ) {
+    ) where
+        C: Risc0Context + ?Sized,
+    {
```

Comment:
I'm wondering why this is needed?

Reply:
This must have been an artifact of different revisions. I've changed it back to an `impl`
bdc75e2802cf3f7a3043c1b84e69ae84eb03f85e